### PR TITLE
Pass require path to sub-process when running `bundle install`

### PR DESCRIPTION
**Pull request by justinplus**
This is a 🐛 bug fix.

## Summary

Pass the require path of `bundler` to the sub-process to ensure it uses the correct version.

## Context
Basically, this small change is covered in the original test.

Resolves: #7616

### CODE REVIEW

**Comment by justinplus** [comment]
I couldn't find which case was failed in the CI's log. Thus I ran the tests in my local and I found this test was failed.

```log
Failure:
TestKramdown#test_: kramdown when a custom highlighter is chosen should use the chosen highlighter if it's available.  [/Users/tjia/Repos/jekyll-fork/test/test_kramdown.rb:119]
Minitest::Assertion: Expected true to not be truthy.

865 tests, 1566 assertions, 1 failures, 0 errors, 0 skips
```

But this seems to be unrelated to my change.

**Comment by mattr-** [comment]
@justinplus Would you be able to look into why the CI has failed?

File: lib/jekyll/commands/new.rb [hunk 1]
```diff
@@ -153,8 +153,11 @@ def after_install(path, options = {})
         def bundle_install(path)
           Jekyll.logger.info "Running bundle install in #{path.cyan}..."
           Dir.chdir(path) do
-            exe = Gem.bin_path("bundler", "bundle")
-            process, output = Jekyll::Utils::Exec.run("ruby", exe, "install")
+            gem "bundler"
```
Comment (ashmaroli) [inline lib/jekyll/commands/new.rb]: What is this line supposed to do..?

Comment (justinplus) [reply lib/jekyll/commands/new.rb]: `gem "bundler"` activates a gem and puts it into the `loaded_specs`.
For more details, please refer to the source code of this method.
https://rubydocs.org/d/ruby-2-6-2/classes/Kernel.html#method-i-gem

Comment (ashmaroli) [reply lib/jekyll/commands/new.rb]: Nice.. I had no idea that private instance methods were documented as well.
Could you please add a comment above this line of code to help future readers refer to the docs as well.
```ruby
# Activate 'bundler' gem and puts it into the `loaded_specs`.
# For details: https://rubydocs.org/d/ruby-2-4-0/classes/Kernel.html#method-i-gem
gem "bundler"
```

**Comment by justinplus** [comment]
After retry, I could successfully run the `cibuild` in my own Mac this time. Are there some unstable test cases?
(by the way, I use Ruby 2.6.2, bundle 2.0.1)
```log
$ git log -n 1 fix-bundle-install 
commit 38eac07a935e7b83e55fbdbb3d965ba5ee636b7c (HEAD -> fix-bundle-install, origin/master, origin/fix-bundle-install, origin/HEAD, master)
Author: Justin Jia <justinj656@live.cn>
Date:   Mon Apr 15 21:00:29 2019 +0800

    Pass require path to sub-process when running `bundle install`
tjia@LM-SHB-24502074 ~/Repos/jekyll-fork
$ script/cibuild
 ---------------------------------------------------------- 
          _   ______   _  __ __     __  _        _          
         | | |  ____| | |/ / \ \   / / | |      | |         
         | | | |__    | ' /   \ \_/ /  | |      | |         
     _   | | |  __|   |  <     \   /   | |      | |         
    | |__| | | |____  | . \     | |    | |____  | |____     
     \____/  |______| |_|\_\    |_|    |______| |______|    
                                                            
 ---------------------------------------------------------- 
Rubocop 0.66.0
Inspecting 129 files
.................................................................................................................................

129 files inspected, no offenses detected
+ ruby -S bundle exec rake TESTOPTS=--profile test
/usr/local/Cellar/ruby/2.6.2/bin/ruby -w -I"lib:lib:test" -I"/usr/local/Cellar/ruby/2.6.2/lib/ruby/gems/2.6.0/gems/rake-12.3.2/lib" "/usr/local/Cellar/ruby/2.6.2/lib/ruby/gems/2.6.0/gems/rake-12.3.2/lib/rake/rake_test_loader.rb" "test/test_ansi.rb" "test/test_cleaner.rb" "test/test_coffeescript.rb" "test/test_collections.rb" "test/test_command.rb" "test/test_commands_serve.rb" "test/test_commands_serve_servlet.rb" "test/test_configuration.rb" "test/test_convertible.rb" "test/test_data_reader.rb" "test/test_doctor_command.rb" "test/test_document.rb" "test/test_drop.rb" "test/test_entry_filter.rb" "test/test_excerpt.rb" "test/test_excerpt_drop.rb" "test/test_filters.rb" "test/test_front_matter_defaults.rb" "test/test_generated_site.rb" "test/test_kramdown.rb" "test/test_layout_reader.rb" "test/test_liquid_extensions.rb" "test/test_liquid_renderer.rb" "test/test_log_adapter.rb" "test/test_new_command.rb" "test/test_page.rb" "test/test_page_without_a_file.rb" "test/test_path_sanitization.rb" "test/test_plugin_manager.rb" "test/test_post_reader.rb" "test/test_regenerator.rb" "test/test_related_posts.rb" "test/test_sass.rb" "test/test_site.rb" "test/test_site_drop.rb" "test/test_static_file.rb" "test/test_tags.rb" "test/test_theme.rb" "test/test_theme_assets_reader.rb" "test/test_url.rb" "test/test_utils.rb" --profile
# -------------------------------------------------------------
# SPECS AND TESTS ARE RUNNING WITH WARNINGS OFF.
# SEE: https://github.com/Shopify/liquid/issues/730
# SEE: https://github.com/jekyll/jekyll/issues/4719
# -------------------------------------------------------------

# Running tests with run options --profile --seed 53446:

.................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................

Finished tests in 93.017210s, 9.2994 tests/s, 16.8356 assertions/s.


865 tests, 1566 assertions, 0 failures, 0 errors, 0 skips

================================================================================
Your 10 Slowest Tests
================================================================================

 3.7757ms - TestSite#test_: creating sites should write static files if not modified but missing in destination. 
 3.7696ms - TestSite#test_: creating sites should write only modified static files. 
 3.6249ms - TestSite#test_: creating sites incremental build should build incrementally. 
 1.4367ms - TestNewCommand#test_: when args contains a path should force created folder. 
 1.3999ms - TestSite#test_: creating sites incremental build should regenerate files that have had their destination deleted. 
 1.0631ms - TestNewCommand#test_: when args contains a path should process any ERB files. 
 1.0098ms - TestStaticFile#test_: A StaticFile should known if the source path is modified, when it is. 
 1.0047ms - TestStaticFile#test_: A StaticFile should known if the source path is modified, when it's not. 
 0.8956ms - TestRegenerator#test_: The site metadata should not regenerate again if multiple dependencies. 
 0.8264ms - TestCoffeeScript#test_: converting CoffeeScript should produce JS. 

Coverage report generated for Unit Tests to /Users/tjia/Repos/jekyll-fork/coverage. 3538 / 3931 LOC (90.0%) covered.

real	1m37.160s
user	0m55.626s
sys	0m25.565s

*** the left is omitted ***
```

It seems the Travis CI job of my PR (https://travis-ci.org/jekyll/jekyll/jobs/520285065) was aborted in the middle and it even didn't show the failed test cases. Could you help to take a look?
Could we retry the CI job first?

**Comment by ashmaroli** [comment]
> It seems the Travis CI job of my PR...

Its not just Travis job that fails, but AppVeyor jobs as well. Anyways, I've restarted the failed jobs. But I doubt they'd pass..

**Comment by justinplus** [comment]
All tests have passed in my local. How can I debug the travis-ci failure?

**Comment by ashmaroli** [comment]
> How can I debug the travis-ci failure?

This is a really puzzling bug:
  - It's not reproducible when running `script/cibuild` locally.
  - It's not reproducible when singularly running `test/test_new_command.rb` in CI or locally.
  - It's not reproducible by running `script/default-site` (tests `jekyll new` command behavior) in CI or locally.

Yet, it **always fails** when `script/test` (the entire unit-test suite) is run in CI...... :confused:

**Comment by justinplus** [comment]
> Yet, it always fails when script/test (the entire unit-test suite) is run in CI...... 😕

Found this stack trace in CI job (https://travis-ci.org/jekyll/jekyll/jobs/521240884)
```log
Traceback (most recent call last):
  11: from /home/travis/build/jekyll/jekyll/vendor/bundle/ruby/2.6.0/gems/simplecov-0.16.1/lib/simplecov/defaults.rb:27:in `block in <top (required)>'
  10: from /home/travis/build/jekyll/jekyll/vendor/bundle/ruby/2.6.0/gems/simplecov-0.16.1/lib/simplecov.rb:200:in `run_exit_tasks!'
   9: from /home/travis/build/jekyll/jekyll/vendor/bundle/ruby/2.6.0/gems/simplecov-0.16.1/lib/simplecov/configuration.rb:182:in `block in at_exit'
   8: from /home/travis/build/jekyll/jekyll/vendor/bundle/ruby/2.6.0/gems/simplecov-0.16.1/lib/simplecov/result.rb:48:in `format!'
   7: from /home/travis/build/jekyll/jekyll/vendor/bundle/ruby/2.6.0/gems/simplecov-html-0.10.2/lib/simplecov-html.rb:22:in `format'
   6: from /home/travis/build/jekyll/jekyll/vendor/bundle/ruby/2.6.0/gems/simplecov-html-0.10.2/lib/simplecov-html.rb:22:in `open'
   5: from /home/travis/build/jekyll/jekyll/vendor/bundle/ruby/2.6.0/gems/simplecov-html-0.10.2/lib/simplecov-html.rb:23:in `block in format'
   4: from /home/travis/.rvm/rubies/ruby-2.6.0/lib/ruby/2.6.0/erb.rb:901:in `result'
   3: from /home/travis/.rvm/rubies/ruby-2.6.0/lib/ruby/2.6.0/erb.rb:901:in `eval'
   2: from (erb):17:in `block in format'
   1: from /home/travis/build/jekyll/jekyll/vendor/bundle/ruby/2.6.0/gems/simplecov-html-0.10.2/lib/simplecov-html.rb:95:in `timeago'
/home/travis/.rvm/rubies/ruby-2.6.0/lib/ruby/2.6.0/time.rb:717:in `xmlschema': can't modify frozen String (FrozenError)
``` 

Do you have any idea?

**Comment by ashmaroli** [comment]
> Do you have any idea?

Unfortunately, no. Like other anomalies, that trace is present only with Ruby 2.6.. :man_shrugging:

**Comment by mattr-** [comment]
Let's turn off coverage reporting for Ruby 2.6. We only need coverage output from a single build anyways.

**Comment by mattr-** [comment]
Since the Actions builds are green even though the others are red, i'll merge this anyways.

@jekyllbot: merge +minor

**Comment by ashmaroli** [comment]
Unfortunately, merging this has broken our test-suite on Rubies older than Ruby 2.7 as well as on JRuby.
There is no error message and therefore cannot reliably ship this change to our users.
Perhaps there's an alternate way to resolve the issue addressed by this PR...?

**Comment by DirtyF** [comment]
This has been reverted for now, until we find a proper solution to resolve this on Ruby > 2.5

**Pull request merged**